pkg/config/piped.go

```diff
@@ -82,6 +84,37 @@ type PipedSpec struct {
 	AppSelector map[string]string `json:"appSelector,omitempty"`
 }
 
+func (s *PipedSpec) UnmarshalJSON(data []byte) error {
+	ps := pipedSpec{}
+	if err := json.Unmarshal(data, &ps); err != nil {
+		return err
+	}
+
+	s.ProjectID = ps.ProjectID
+	s.PipedID = ps.PipedID
+	s.PipedKeyFile = ps.PipedKeyFile
+	s.PipedKeyData = ps.PipedKeyData
+	s.Name = ps.Name
+	s.APIAddress = ps.APIAddress
+	s.WebAddress = ps.WebAddress
+	s.SyncInterval = ps.SyncInterval
+	s.AppConfigSyncInterval = ps.AppConfigSyncInterval
+	s.Git = ps.Git
+	s.Repositories = ps.Repositories
+	s.ChartRegistries = ps.ChartRegistries
+	s.ChartRepositories = ps.ChartRepositories
+	// Add all CloudProviders configuration as PlatformProviders configuration.
+	s.PlatformProviders = append(s.PlatformProviders, ps.PlatformProviders...)
+	s.PlatformProviders = append(s.PlatformProviders, ps.CloudProviders...)
+	s.AnalysisProviders = ps.AnalysisProviders
+	s.Notifications = ps.Notifications
+	s.SecretManagement = ps.SecretManagement
+	s.EventWatcher = ps.EventWatcher
+	s.AppSelector = ps.AppSelector
+
+	return nil
+}
```

Comment:
We should avoid this manual copy.
How about this?

```suggestion
func (s *PipedSpec) UnmarshalJSON(data []byte) error {
        type Alias PipedSpec

        ps := &struct {
                *Alias
        }{
                Alias: (*Alias)(s),
        }
        if err := json.Unmarshal(b, &ps); err != nil {
                return err
        }
    
	// Add all CloudProviders configuration as PlatformProviders configuration.
	s.PlatformProviders = append(s.PlatformProviders, ps.CloudProviders...)
	s.CloudProviders = nil

	return nil
}
```

Reply:
Nice, thanks to teach me about that 🙏

Reply:
addressed by https://github.com/pipe-cd/pipecd/pull/3815/commits/a7a47672c3098639458791237f6c22d9d550cbe8